src/property-filter/__tests__/property-filter-enum-tokens.test.tsx

```diff
@@ -0,0 +1,550 @@
+  });
+
+  test('shows filtering empty if no enum token values are provided', () => {
+    const { wrapper } = renderComponent({ filteringOptions: [] });
+
+    wrapper.focus();
+    wrapper.setInputValue('state =');
+
+    expect(wrapper.findStatusIndicator()!.getElement()).toHaveTextContent('Empty');
+  });
+
+  test('shows filtering empty if no enum token values are matched', () => {
+    const { wrapper } = renderComponent();
+
+    wrapper.focus();
+    wrapper.setInputValue('state = X');
+
+    expect(wrapper.findStatusIndicator()!.getElement()).toHaveTextContent('Empty');
+  });
+});
+
+describe('sync token editing', () => {
+  test('can replace value token with enum token', () => {
+    const { openEditor, findEditor, getTokensContent, onChange } = renderComponent({
+      query: { operation: 'and', tokens: [{ propertyKey: 'state', operator: ':', value: '0' }] },
+    });
+
+    openEditor();
+    findEditor().findOperatorField().findControl()!.findSelect()!.openDropdown();
+    findEditor().findOperatorField().findControl()!.findSelect()!.selectOptionByValue('=');
+    findEditor().findSubmitButton().click();
+
+    expect(getTokensContent()).toEqual(['State = ']);
+    expect(onChange).toHaveBeenCalledWith(
+      expect.objectContaining({
+        detail: { operation: 'and', tokens: [{ propertyKey: 'state', operator: '=', value: [] }] },
+      })
+    );
+  });
+
+  test('can replace enum token with value token', () => {
+    const { openEditor, findEditor, getTokensContent, onChange } = renderComponent({
+      query: { operation: 'and', tokens: [{ propertyKey: 'state', operator: '=', value: ['0'] }] },
+    });
+
+    openEditor();
+    findEditor().findOperatorField().findControl()!.findSelect()!.openDropdown();
+    findEditor().findOperatorField().findControl()!.findSelect()!.selectOptionByValue(':');
+    findEditor().findSubmitButton().click();
+
+    expect(getTokensContent()).toEqual(['State : ']);
+    expect(onChange).toHaveBeenCalledWith(
+      expect.objectContaining({
+        detail: { operation: 'and', tokens: [{ propertyKey: 'state', operator: ':', value: null }] },
+      })
+    );
+  });
+
+  test('can change value in enum token', () => {
+    const { openEditor, findEditor, findEditorMultiselect, getTokensContent, onChange } = renderComponent({
+      query: { operation: 'and', tokens: [{ propertyKey: 'state', operator: '=', value: ['0'] }] },
+    });
+
+    openEditor();
+    findEditorMultiselect().openDropdown();
+    findEditorMultiselect().selectOptionByValue('1');
+    findEditor().findSubmitButton().click();
+
+    expect(getTokensContent()).toEqual(['State = Stopped, Stopping']);
+    expect(onChange).toHaveBeenCalledWith(
+      expect.objectContaining({
+        detail: { operation: 'and', tokens: [{ propertyKey: 'state', operator: '=', value: ['0', '1'] }] },
+      })
+    );
+  });
+
+  test('keeps value when changing operator so that value type preserves', () => {
```

Comment:
Why not also cover the complementary case: that the value is cleared when switching from an enum-type to a value-type operator and vice versa?

Reply:
Oh I see, this is already done further up in `can replace value token with enum token` and `can replace enum token with value token`. My bad